cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -40,11 +40,40 @@ type MasterConfiguration struct {
 	KubernetesVersion string
 	CloudProvider     string
 	AuthorizationMode string
+
+	Phases Phases
+
+	// The directory where certificates are stored
+	CertificatesDir string
+}
+
+type PhaseMetadata struct {
+	Annotations map[string]string
+}
+
+type Phases struct {
+	Certificates CertificatesPhase
```

Comment:
I considered that, but I started thinking if it's better to have the options (ie. SelfSign) as pointers instead... WDYT?